### DISCUSSION TITLE
Migrations vs Schemas, what is the way to go?

### DISCUSSION THREAD

**Discussion by gabrsar** [directus/directus#12713] [Archive]
Hi everyone.

From what I could grasp there is a gap in the docs, about schemas and migrations and which one should I use for each task. Looks like there is an overlap in the responsibility of fields manipulation (DDL) between them.

From my experiences and what I saw on the docs and a bit of directus source code, I found that you can create/drop fields simply by adding them to the schema and importing it. Also, you can do it using migrations too.

There is any way of disabling schema`s fields & relations handling and only letting it handle the UI & soft rules, while DDL changes through migrations?

I'm looking for a clean way to promote changes I made on my local/test env to the production environment (both new fields and their configs):

For example:

In this order:
add/remove/change fields and tables -> run migrations
customize field appearance, who can access it, ... -> import a new version of the schema

My main concern is forgetting to update or messing with the schema and it ends up dropping some fields in production.

Is my rationale aligned with the Directus premisses?

Thank you very much.

**Comment by rijkvanzanten** [comment]
Heya! Sounds like schema migrations is what you'd be looking for. Those allow you to create a snapshot of the current configuration, and apply that to another instance. Migrations could also be used for this purpose, but are intended for more edge-case uses (it's an escape hatch in case schema migrations doesn't offer what you need) 👍🏻
